openmc/mgxs/mdgxs.py

```diff
@@ -388,19 +410,53 @@ def get_xs(self, groups='all', subdomains='all', nuclides='all',
 
         # Reshape tally data array with separate axes for domain, energy groups,
         # delayed groups, and nuclides
-        num_subdomains = int(xs.shape[0] / (num_groups * num_delayed_groups))
-        new_shape = (num_subdomains, num_delayed_groups, num_groups)
-        new_shape += xs.shape[1:]
-        xs = np.reshape(xs, new_shape)
+        # Accomodate the polar and azimuthal bins if needed
+        if self.num_polar or self.num_azimuthal:
```

Comment:
The `num_pol` and `num_azi` were used to store the results of the if-block (that you point out can be removed).  So in that sense, they are needed; if I get rid of the need for the if-block, then you're right they are unnecessary